simtools/utils/general.py

```diff
@@ -917,12 +917,13 @@ def get_value_unit_type(value, unit_str=None):
         base_type = extract_type_of_value(base_value)
 
     if unit_str is not None:
-        if base_unit is not None:
-            try:
-                base_value = base_value * u.Unit(base_unit).to(u.Unit(unit_str))
-            except u.UnitConversionError:
-                _logger.error(f"Cannot convert {base_unit} to {unit_str}.")
-                raise
+        try:
+            base_value = base_value * u.Unit(base_unit).to(u.Unit(unit_str))
+        except u.UnitConversionError:
+            _logger.error(f"Cannot convert {base_unit} to {unit_str}.")
+            raise
+        except TypeError:
```

Comment:
it is not clear to me when the TypeError occurs? If base_unit is None?

We are still not checking whether unit_str makes sense before proceeding. Passing None or "" to `unit_str` leads to None being returned as base_unit.

Reply:
Exactly, u.Unit(Unit) gives you a TypeError.

And yes, returning None is also ok for when there is no base unit. This is intended (also in the unit tests)